xwiki-platform-core/xwiki-platform-localization/xwiki-platform-localization-rest/src/main/java/org/xwiki/localization/internal/DefaultLocalizationSource.java

```diff
@@ -0,0 +1,117 @@
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.localization.internal;
+
+import java.util.List;
+import java.util.Locale;
+import java.util.Objects;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Provider;
+import javax.inject.Singleton;
+import javax.ws.rs.core.Response;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.xwiki.component.annotation.Component;
+import org.xwiki.component.manager.ComponentLookupException;
+import org.xwiki.component.manager.ComponentManager;
+import org.xwiki.localization.LocalizationContext;
+import org.xwiki.localization.LocalizationManager;
+import org.xwiki.localization.Translation;
+import org.xwiki.localization.rest.LocalizationSource;
+import org.xwiki.rest.XWikiRestComponent;
+
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.xpn.xwiki.XWikiContext;
+
+/**
+ * Default implementation of {@link LocalizationSource}.
+ *
+ * @version $Id$
+ * @since 13.3RC1
+ */
+@Component
+@Named("org.xwiki.localization.internal.DefaultLocalizationSource")
+@Singleton
+public class DefaultLocalizationSource implements LocalizationSource, XWikiRestComponent
+{
+    @Inject
+    @Named("context")
+    private Provider<ComponentManager> componentManagerProvider;
+
+    @Inject
+    private Provider<XWikiContext> contextProvider;
+
+    @Inject
+    private Logger logger;
+
+    @Override
+    public Response translations(String wikiId, String locale, String prefix, List<String> keys)
+    {
+        XWikiContext xWikiContext = this.contextProvider.get();
+
+        // Save the current context wiki id. It will be restored on the method is finished.
+        String oldWikiId = xWikiContext.getWikiId();
+        try {
+            // Set the requested wiki id.
+            xWikiContext.setWikiId(wikiId);
+
+            ComponentManager componentManager = this.componentManagerProvider.get();
+            LocalizationContext localizationContext = componentManager.getInstance(LocalizationContext.class);
+            LocalizationManager localizationManager = componentManager.getInstance(LocalizationManager.class);
+
+            // Resolve the local.
+            Locale local;
+            if (StringUtils.isNotEmpty(locale)) {
+                local = Locale.forLanguageTag(locale);
```

Comment:
You should use `org.xwiki.localization.LocaleUtils#toLocale` instead.

Reply:
Good catch, thanks.

Reply:
https://github.com/xwiki/xwiki-platform/pull/1595/files#diff-118a400f3886da870ebde91eda0d6269f44a19975059f7d189964ac02c48d623R82